### PR TITLE
fix: support ext convention

### PR DESCRIPTION
**Pull request by nobuhikosawai**
I saw this https://github.com/nvim-tree/nvim-web-devicons/pull/191 and I like the idea of supporting `*.test.js`, `*.stories.tsx`, etc because these are quite common conventions.

The problem is currently this is not working as intended. (Only `test.js` highlights but not `foo.test.js`)

image

This PR is to fix this issue.
image

*Note*
Whether a plugin can reflect this change depends on the plugin's implementation.
(I see some attempt to support option to switch the way to get icons like this PR: https://github.com/akinsho/bufferline.nvim/pull/669)

image

### CODE REVIEW

File: lua/nvim-web-devicons.lua [hunk 1]
```diff
@@ -1736,7 +1736,7 @@ local function get_default_icon()
 end
 
 local function get_icon(name, ext, opts)
-  ext = ext or name:match("^.*%.(.*)$") or ""
+  ext = ext or name:match("^.*%.(.*%..*)$") or name:match("^.*%.(.*)$") or ""
```
Comment (alex-courtis) [inline lua/nvim-web-devicons.lua:1739]: Could we simply match `"^.*%.(.*%..*)$"` without `"^.*%.(.*)$"`? Greedy match should cover all existing cases...

**Review by alex-courtis — CHANGES_REQUESTED** [review]
Your timing is excellent; I've just been testing #197

This one's a bug, not a feature and doesn't break anything.

Tested OK for cases you referenced, as well as regular cases.

We'll merge that one first.

File: lua/nvim-web-devicons.lua (hunk 1 above)
Comment (nobuhikosawai) [reply lua/nvim-web-devicons.lua]: @alex-courtis Thanks for your feedback. I've updated the regexp pattern to match both 'foo.test.js' and 'foo.js'. Could you double check?

**Review by alex-courtis — APPROVED** [review]
Tested OK:

bleh.stories.tsx
bleh.tsx
stories.tsx

Will merge #197 first

**Review by alex-courtis — APPROVED** [review]
Merged master, retested OK.

Many thanks for your contribution.

**Comment by hthuong09** [comment]
But then again it breaks other types.

For example, I have a file called `cat.service.ts`, the extension is now `service.ts` and it does not match any icon.

I would suggest adding another variable to this case and checking it first before the actual extension.

**Comment by alex-courtis** [comment]
Reported #205

Reverting.

**Pull request merged**